tests/acceptance/builds/prioritize-test.js

```diff
@@ -0,0 +1,84 @@
+    signInUser(currentUser);
+  });
+
+  test('prioritizing build button does not exsist if the build is not having permission to prioritize becuase of the organization', async function (assert) {
+    let repository =  this.server.create('repository', { slug: 'travis-ci/travis-web' });
+    let branch = this.server.create('branch', { repository, name: 'acceptance-tests', default_branch: true });
+    let gitUser = this.server.create('git-user', { name: 'Mr T' });
+    let commit = this.server.create('commit', { author: gitUser, committer: gitUser, committer_name: 'Mr T', message: 'This is a message' });
+    let build = this.server.create('build', { number: '5', state: 'queued', permissions: { 'prioritize': false }, repository, commit, branch, priority: true, });
+    let job = this.server.create('job', { number: '1234.1', state: 'queued', repository, commit, build });
+    this.server.create('log', { id: job.id });
+
+    await visit('/travis-ci/travis-web');
+    assert.dom('[data-test-repo-actions-prioritize-button]').doesNotExist();
+  });
+
+  test('prioritizing build button does not exsist if the build is in the finished states', async function (assert) {
+    let repository =  this.server.create('repository', { slug: 'travis-ci/travis-web' });
+    let branch = this.server.create('branch', { repository, name: 'acceptance-tests', default_branch: true });
+    let gitUser = this.server.create('git-user', { name: 'Mr T' });
+    let commit = this.server.create('commit', { author: gitUser, committer: gitUser, committer_name: 'Mr T', message: 'This is a message' });
+    let build = this.server.create('build', { number: '5', state: 'passed', permissions: { 'prioritize': false }, repository, commit, branch, priority: true, });
+    let job = this.server.create('job', { number: '1234.1', state: 'passed', repository, commit, build });
+    this.server.create('log', { id: job.id });
+
+    await visit('/travis-ci/travis-web');
+    assert.dom('[data-test-repo-actions-prioritize-button]').doesNotExist();
+  });
+
+  test('prioritizing build with favouorable conditions', async function (assert) {
+    let repository =  this.server.create('repository', { slug: 'travis-ci/travis-web' });
+    let branch = this.server.create('branch', { repository, name: 'acceptance-tests', default_branch: true });
+    let gitUser = this.server.create('git-user', { name: 'Mr T' });
+    let commit = this.server.create('commit', { author: gitUser, committer: gitUser, committer_name: 'Mr T', message: 'This is a message' });
+    let build = this.server.create('build', { number: '5', state: 'queued', permissions: { 'prioritize': true }, repository, commit, branch, priority: true, });
+    let job = this.server.create('job', { number: '1234.1', state: 'queued', repository, commit, build });
+    this.server.create('log', { id: job.id });
+
+    await visit('/travis-ci/travis-web');
+    assert.dom('[data-test-repo-actions-prioritize-button]').exists();
```

Comment:
I'm curious, why the button exists here when the build is already high priority (assuming based on `build.priority: true`)

Reply:
As per line 60, the build priority is false, which means it is not prioritized yet, hence the button is visible. Rest I have checked this point onto the application by reversing this condition and the button is not coming which is the expected behavior.